### ISSUE TITLE
Desync in multiplayer if someone rotates the view

### ISSUE THREAD

**Issue by Buchstabensalat12** [OpenRCT2/OpenRCT2#20294]
### Operating System

Windows 10 64bit

### OpenRCT2 build

OpenRCT2, v0.4.5-49-g93c1a97 (93c1a97 on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This bug is specific to multiplayer

### Describe the issue

When playing in multiplayer, if someone rotates the view, the clients are getting a desync.
It is not important who rotates his view (Server or client).

### Steps to reproduce

1. Host a multiplayer game (We did play Extreme Heights of the RCT2 base game)
2. Connect a another player as client
3. Either the server player or the client rotates his view
4. Desync at the client

**Comment by Broxzier** [comment]
I cannot reproduce this issue with the steps you provided. Did one of you perhaps use a plug-in that could interfere with the game logic that somehow gets triggered when the camera rotates?

**Comment by ZehMatt** [comment]
Also tested this and I can't reproduce this issue on my end. also tried headless and same result.

**Comment by Buchstabensalat12** [comment]
We both had one plugin installed:
- The server had "AdvancedTracks" installed
- I, the client had "Minesweeper" installed

We both removed the plugins, restarted both the game, but the desync still happens

**Comment by ZehMatt** [comment]
Can you provide the park you used?

**Comment by Buchstabensalat12** [comment]
Here is the savegame, it is a recently started save of the park "Extreme Heights" from the RCT2 basegame:

[Extreme_Hohen_MP.zip](https://github.com/OpenRCT2/OpenRCT2/files/11578703/Extreme_Hohen_MP.zip)

**Comment by Buchstabensalat12** [comment]
I did some tests today with my PC and my laptop as the second device.
I used the current develop version on both devices: OpenRCT2, v0.4.5-55-g29bd730 (29bd730 on develop) provided by GitHub
On both devices the game is started via Open Launcher.

Both devices are connected via gigabit ethernet on the same ethernet switch (for ruling out internet or WiFi isseues).

I started the server with a new park on my PC (Crazy Castle, RCT2 basegame) and connect with my laptop to the server.
The desync is still happening when rotating the view, but sometimes I can rotate one time without desyncing and the second rotation causes the desync.

I think there the issue is related to my setup, else more people would suffer from this issue,

The specs of both machines:
PC:
- CPU: AMD Ryzen 7 3700X
- RAM: 32GB DDR4 3200
- GPU: Nvidia Geforce RTX 2070 Super
- OS: Windows 10, 64 Bit

Laptop:
- CPU: AMD Ryzen 5 5600H
- RAM: 16GB DDR4 3200
- GPU1: Integrated Radeon graphics (used for OpenRCT2)
- GPU2: Nvidia Geforce RTX 3060 Laptop GPU (115W version)
- OS: Windows 10, 64 Bit

I also attached the OpenRCT2 config file of both machines: [configfiles.zip](https://github.com/OpenRCT2/OpenRCT2/files/11584012/configfiles.zip)

One interesting thing I have seen at my test, is that OpenRCT2 reports a ping jumping between 25ms and 75ms while the ping tool of Windows reports a ping of less than one ms.
![Latency](https://github.com/OpenRCT2/OpenRCT2/assets/7858087/2ede0d2a-0a83-4910-8001-4f73a1df0e7c)
(sorry for the german at the screenshot, I don't know if it is possible to switch the language of Windows on the running system)

Is there any way to give you more information about the desync?
I see a folder called "desyncs" in the OpenRCT2 folder but it is empty.

**Comment by ZehMatt** [comment]
Can you turn desync_debugging to true in the server config, the client will generate some files in OpenRCT2\desyncs folder when a desync happens, those might give some insight to whats happening, I can not reproduce this on my machine.

**Comment by Buchstabensalat12** [comment]
I set desync_debugging to true and this was the resulting log:
[desync_18075131044111151624_1498.txt](https://github.com/OpenRCT2/OpenRCT2/files/11589741/desync_18075131044111151624_1498.txt)

I started the server (PC) with the park "Crazy Castle" connected the client (laptop) to the server. The desync happend when I rotated the view at the client.


After that I restarted the server and did the same, the only thing I did different was that I rotated the view on the server. Here is the desync log: 
[desync_18075131048702467416_1914.txt](https://github.com/OpenRCT2/OpenRCT2/files/11589793/desync_18075131048702467416_1914.txt)

I hope this information helps to find the the cause of the desync.

Yesterday I did another test: My PC is a dual boot system, where the second OS is Linux Debian 12 "Bookworm" (currently testing, but in less than two weeks the new stable).
So I installed the matching Linux AppImage version of OpenRCT2 (the version downloaded by the Linux version of OpenLauncher doesn't work) and tried this as the server.
But even there I got the desync.

**Comment by ZehMatt** [comment]
Out of curiosity, do you get the desyncs when you host and join from the same computer? No matter what I do here I can't reproduce it, I went over all the various settings that you kindly provided and nothing changed it. Also are you running on both machines the same OS?

**Comment by Buchstabensalat12** [comment]
The desync still happens if I run two instances of OpenRCT2 on the same PC.


I did some testing with options and the issue seems to be **related to framerate**.
I usually play with uncapped framerate and without V-Sync (due to having a G-SYNC capable monitor).

If I activate the 40 fps framerate cap (uncap_fps = false), then the desync doesn't happen anymore.
Enabling / disabling multithreading or switching between software rendering and OpenGL doesn't have any effect on the desync.

If the server is running with uncapped framerate and the server player rotates his view, the desync happens even if the client is running at 40fps cap.
If the client rotates his view and has uncapped framerate, then he desyncs regardless of the framerate at the server.

When I activate V-Sync to limit the framerate to the speed of the monitor (144fps at my PC, 120fps at my laptop) the desync still happens.

**Comment by ZehMatt** [comment]
Quite the curious case for sure, technically speaking this should be impossible to happen as the view rotation is not stored on any entity and the view rotation has typically no influence in the simulation, vsync and uncap fps also should have no influence, internally it runs always at 40Hz, uncap fps just means it renders more. I'm not sure how to investigate this issue for as long I can't reproduce this issue.

**Comment by Buchstabensalat12** [comment]
I downloaded several of the recent development builds to check if I can find the version where the desyncs start.

I found that "OpenRCT2, v0.4.5-48-g66ae20b (66ae20b on develop) provided by GitHub" is the first version, where the desync does happen.

At the previous development version (OpenRCT2, v0.4.5-45-gf041680 (f041680 on develop) provided by GitHub), I don't have that problem.

I hope this helps to find the cause of the problem.
